test/suite/test_prepare_lookaside02.py

```diff
@@ -0,0 +1,174 @@
+            self.skipTest('requires a timestamp build')
+
+        self.session.create(self.uri,
+            'key_format=i,value_format=i' + self.extra_config)
+        c = self.session.open_cursor(self.uri)
+
+        # Insert keys 1..100 each with timestamp=key, in some order
+        orig_keys = range(1, 101)
+        keys = orig_keys[:]
+        random.shuffle(keys)
+
+        # Scenario: 1
+        # Check insert operation
+        self.session.begin_transaction(self.txn_config)
+        c[1] = 1
+        # update the value with in this txn
+        self.session.prepare_transaction('prepare_timestamp=' + timestamp_str(100))
+        if self.txn_commit == True:
+            self.session.commit_transaction('commit_timestamp=' + timestamp_str(101))
+        else:
+            self.session.rollback_transaction()
+
+        self.conn.set_timestamp('stable_timestamp=' + timestamp_str(150))
+        self.conn.set_timestamp('oldest_timestamp=' + timestamp_str(150))
+        self.session.checkpoint()
+
+        # Scenario: 2
+        # Check update operation
+        #   update a existing key.
+        #   update a newly inserted key with in this transaction
+        self.session.begin_transaction(self.txn_config)
+        # update a committed value, key 1 is inserted above.
+        c[1] = 2
+        # update a uncommitted value, insert and update a key.
+        c[2] = 1
+        c[2] = 2
+        self.session.prepare_transaction('prepare_timestamp=' + timestamp_str(200))
+        if self.txn_commit == True:
+            self.session.commit_transaction('commit_timestamp=' + timestamp_str(201))
+        else:
+            self.session.rollback_transaction()
+
+        self.conn.set_timestamp('stable_timestamp=' + timestamp_str(250))
+        self.conn.set_timestamp('oldest_timestamp=' + timestamp_str(250))
+        self.session.checkpoint()
+
+        # Scenario: 3
+        # Check remove operation
+        #   remove an existing key.
+        #   remove an previously updated key.
+        #   remove a newly inserted and updated key.
+        self.session.begin_transaction(self.txn_config)
+        # update a committed value, key 1 is inserted above.
+        c.set_key(1)
+        c.remove()
+        c.set_key(2)
+        c.remove()
+        c[3] = 1
+        c[3] = 2
+        c.set_key(3)
+        c.remove()
+        self.session.prepare_transaction('prepare_timestamp=' + timestamp_str(300))
+        if self.txn_commit == True:
+            self.session.commit_transaction('commit_timestamp=' + timestamp_str(301))
+        else:
+            self.session.rollback_transaction()
+
+        # commit some keys, to generate the update chain subsequently.
+        self.session.begin_transaction(self.txn_config)
+        c[1] = 1
+        c[2] = 1
+        c[3] = 1
+        self.session.commit_transaction('commit_timestamp=' + timestamp_str(301))
+
+        self.conn.set_timestamp('stable_timestamp=' + timestamp_str(350))
```

Comment:
Similar to my comment before, do we need to advance the timestamp ?
We probably do not need a checkpoint here, connection reopen should take one.